pkg/agent/openflow/pipeline.go

```diff
@@ -894,14 +883,17 @@ func (c *client) l3FwdFlowToRemote(
 func (c *client) l3FwdFlowToRemoteViaGW(
 	localGatewayMAC net.HardwareAddr,
 	peerSubnet net.IPNet,
+	gwOFPort uint32,
 	category cookie.Category) binding.Flow {
 	ipProto := getIPProtocol(peerSubnet.IP)
 	l3FwdTable := c.pipeline[l3ForwardingTable]
 	return l3FwdTable.BuildFlow(priorityNormal).MatchProtocol(ipProto).
 		MatchDstIPNet(peerSubnet).
-		Action().DecTTL().
 		Action().SetDstMAC(localGatewayMAC).
-		Action().GotoTable(l3FwdTable.GetNext()).
+		Action().LoadRegRange(int(PortCacheReg), gwOFPort, ofPortRegRange).
+		// Set MAC-known.
+		Action().LoadRegRange(int(marksReg), portFoundMark, ofPortMarkRange).
+		Action().GotoTable(l3DecTTLTable).
```

Comment:
This is for noEncap mode, right? I think the TTL shouldn't be decremented as the host will do it too.

Reply:
You are right, it is for noEncap case. So you mean here we should let the packet go to L2ForwardCalcTable directly without dec_ttl?

Reply:
I would prefer to let the packet go to ConntrackCommitTable to bypass Ingress rules tables. Since I have changed the flow actions to set target ofport, we could don't need to enforce it to go into L2ForwardingCalcTable. What do you think? @tnqn